composer.json

```diff
@@ -20,33 +20,47 @@
     ],
     "require": {
         "php": "^7.2",
+        "ext-dom": "*",
         "ext-exif": "*",
         "ext-fileinfo": "*",
+        "ext-filter": "*",
         "ext-gd": "*",
+        "ext-hash": "*",
+        "ext-json": "*",
         "behat/transliterator": "^1.1",
         "doctrine/collections": "^1.3",
         "doctrine/data-fixtures": "^1.1",
+        "doctrine/dbal": "^2.9",
         "doctrine/doctrine-bundle": "^1.6",
         "doctrine/doctrine-cache-bundle": "^1.3",
         "doctrine/doctrine-fixtures-bundle": "^2.3",
         "doctrine/doctrine-migrations-bundle": "^2.0",
+        "doctrine/event-manager": "^1.0",
         "doctrine/orm": "^2.5",
+        "doctrine/persistence": "^1.1",
         "egulias/email-validator": "~2.0",
         "friendsofsymfony/oauth-server-bundle": "^1.6",
         "friendsofsymfony/rest-bundle": "^2.1",
         "fzaninotto/faker": "^1.6",
         "gedmo/doctrine-extensions": "^2.4.12",
+        "guzzlehttp/guzzle": "^6.3",
+        "guzzlehttp/psr7": "^1.6",
+        "hwi/oauth-bundle": "^0.6",
```

Comment:
Wondering about this one - we only provide an optional integration with it, it shouldn't be required by all the consumers (that's why it was in require-dev).

Reply:
Do you mean only `hwi/oauth-bundle` and not also `guzzlehttp` packages, right?
If so, yes it's true, `hwi/oauth-bundle` is an optional dependency. I have to think about how to handle it. I think that the right thing to do here would be move this oauth feature in a separate Sylius bundle/plugin where the `hwi/oauth-bundle` is a direct dependency. But it's too much work and should be done in a separate PR.

Reply:
Basically I fully agree with @leonboot here https://github.com/Sylius/Sylius/issues/10530#issuecomment-515063529.
So @pamil I think that we should choose:

1. Move the `Sylius\Bundle\CoreBundle\OAuth\UserProvider` to an external package
2. Require `hwi/oauth-bundle` as strict/direct depencency of Sylius

I'd choose for the option 2. I understand that is an optional integration but I think that requiring `hwi/oauth-bundle` solves more problems than it get.

Reply:
I would suggest adding it to `suggest` and maybe be more clear about this dependency in docs. This is another library that didn't match a stable release. May also block us from Symfony 5.0 support. Less hard dependencies we have, easier it is to maintain the project.

But it is also a question to @CoderMaggie if we want to support login from 3 party apps by default.